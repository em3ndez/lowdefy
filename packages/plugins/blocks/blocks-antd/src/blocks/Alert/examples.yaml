# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Alert
- id: properties.message
  type: Alert
  properties:
    message: Alert message - Completed!
- id: areas.action and properties.message
  type: Alert
  properties:
    message: Alert message - Completed!
  areas:
    action:
      blocks:
        - id: ex_1_block_1
          type: Test
- id: 'properties.icon string'
  type: Info
  properties:
    icon: AiOutlineCheckCircle
    message: Message with Icon!
- id: 'properties.icon object'
  type: Info
  properties:
    icon: # FIX: color and style is not working, className is not passed to alert icon
      name: AiOutlineSetting
      spin: true
      color: red
    message: Message with Icon!
- id: 'properties.type: success'
  type: Alert
  properties:
    message: Alert message - success!
    type: success
- id: 'properties.type: info'
  type: Alert
  properties:
    message: Alert message - info!
    type: info
- id: 'properties.type: warning'
  type: Alert
  properties:
    message: Alert message - warning!
    type: warning
- id: 'properties.type: error'
  type: Alert
  properties:
    message: Alert message - error!
    type: error
- id: 'properties.type: success showIcon: false'
  type: Alert
  properties:
    message: Alert message - success!
    type: success
    showIcon: false
- id: 'properties.type: info showIcon: false'
  type: Alert
  properties:
    message: Alert message - info!
    type: info
    showIcon: false
- id: 'properties.type: warning showIcon: false'
  type: Alert
  properties:
    message: Alert message - warning!
    type: warning
    showIcon: false
- id: 'properties.type: error showIcon: false'
  type: Alert
  properties:
    message: Alert message - error!
    type: error
    showIcon: false
- id: 'properties.type: success banner: true'
  type: Alert
  properties:
    banner: true
    message: Alert message - success!
    type: success
- id: 'properties.type: info banner: true'
  type: Alert
  properties:
    banner: true
    message: Alert message - info!
    type: info
- id: 'properties.type: warning banner: true'
  type: Alert
  properties:
    banner: true
    message: Alert message - warning!
    type: warning
- id: 'properties.type: error banner: true'
  type: Alert
  properties:
    banner: true
    message: Alert message - error!
    type: error
- id: 'properties.closable: true'
  type: info
  properties:
    closable: true
    message: Message closable true
- id: 'properties.closeText: Close Text'
  type: info
  properties:
    closable: true
    closeText: Close Text
    message: Message closable true
- id: with description only
  type: Alert
  properties:
    description: This is a description for the block.
- id: with description only and icon false
  type: Alert
  properties:
    showIcon: false
    description: This is a description for the block.
- id: with description properties.message
  type: Alert
  properties:
    message: Alert message - Completed!
    description: This is a description for the block.
- id: 'with description properties.icon string'
  type: Info
  properties:
    description: This is a description for the block.
    icon: AiOutlineCheckCircle
    message: Message with Icon!
- id: 'with description properties.icon object'
  type: Info
  properties:
    description: This is a description for the block.
    icon: # FIX: color and style is not working, className is not passed to alert icon
      name: AiOutlineSetting
      spin: true
      color: red
    message: Message with Icon!
- id: 'with description properties.type: success'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - success!
    type: success
- id: 'with description properties.type: info'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - info!
    type: info
- id: 'with description properties.type: warning'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - warning!
    type: warning
- id: 'with description properties.type: error'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - error!
    type: error
- id: areas.action and properties.description
  type: Alert
  properties:
    message: Alert message - Completed!
    description: This is a description for the block.
  areas:
    action:
      blocks:
        - id: ex_1_block_1
          type: Test
- id: 'with description properties.type: success showIcon: false'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - success!
    type: success
    showIcon: false
- id: 'with description properties.type: info showIcon: false'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - info!
    type: info
    showIcon: false
- id: 'with description properties.type: warning showIcon: false'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - warning!
    type: warning
    showIcon: false
- id: 'with description properties.type: error showIcon: false'
  type: Alert
  properties:
    description: This is a description for the block.
    message: Alert message - error!
    type: error
    showIcon: false
- id: 'with description properties.type: success banner: true'
  type: Alert
  properties:
    description: This is a description for the block.
    banner: true
    message: Alert message - success!
    type: success
- id: 'with description properties.type: info banner: true'
  type: Alert
  properties:
    description: This is a description for the block.
    banner: true
    message: Alert message - info!
    type: info
- id: 'with description properties.type: warning banner: true'
  type: Alert
  properties:
    description: This is a description for the block.
    banner: true
    message: Alert message - warning!
    type: warning
- id: 'with description properties.type: error banner: true'
  type: Alert
  properties:
    description: This is a description for the block.
    banner: true
    message: Alert message - error!
    type: error
- id: 'with description properties.closable: true'
  type: info
  properties:
    description: This is a description for the block.
    closable: true
    message: Message closable true
- id: 'with description properties.closeText: Close Text'
  type: info
  properties:
    description: This is a description for the block.
    closable: true
    closeText: Close Text
    message: Message closable true
- id: 'with message uses html'
  type: Alert
  properties:
    message: <div style="background:green;">Alert Message</div>
- id: 'with description uses html'
  type: Alert
  properties:
    description: <div style="background:red;">Alert Description</div>
