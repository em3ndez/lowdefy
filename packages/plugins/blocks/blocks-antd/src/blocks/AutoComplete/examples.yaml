# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: AutoComplete
- id: 'properties.options-string'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
- id: properties.title
  type: AutoComplete
  properties:
    title: 'properties.title: My AutoComplete Title'
- id: 'properties.bordered: false'
  type: AutoComplete
  properties:
    bordered: false
- id: 'properties.allowClear: false'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    allowClear: false
- id: 'properties.inputStyle: CSS style applied'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    inputStyle:
      border: 1px solid red
- id: 'properties.optionsStyle: CSS style applied'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    optionsStyle:
      color: blue
- id: 'properties.options-object'
  type: AutoComplete
  properties:
    options:
      - label: one
        value: 'abc'
      - label: two
        value: 'def'
- id: 'properties.autoFocus: true'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    autoFocus: true
- id: 'properties.backfill: true'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    backfill: true
- id: 'properties.defaultOpen: true'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    defaultOpen: true
- id: 'properties.disabled: true'
  type: AutoComplete
  properties:
    disabled: true
- id: 'properties.placeholder_Text_inside_block'
  type: AutoComplete"
  properties:
    placeholder: Text inside block
- id: 'properties.size: small'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    size: small
- id: 'properties.size: large'
  type: AutoComplete
  properties:
    options:
      - Option 1
      - Option 2
    size: large
- id: properties.label.span_12
  type: AutoComplete
  properties:
    label:
      colon: false
      span: 4
- id: properties.label.align_right
  type: AutoComplete
  properties:
    label:
      colon: false
      align: right
- id: properties.label.inline_true
  type: AutoComplete
  properties:
    label:
      colon: false
      inline: true
- id: properties.label.disabled_true
  type: AutoComplete
  properties:
    label:
      colon: false
      disabled: true
- id: properties.label.colon_false
  type: AutoComplete
  properties:
    label:
      colon: false
- id: properties.label.extra_showingextra
  type: AutoComplete
  properties:
    label:
      colon: false
      extra: showing extra
- id: 'properties.options: html'
  type: AutoComplete
  properties:
    options:
      - '<div>Some main text</div><div style="font-size: 6px;">Some small subtext</div>'
      - '<div style="color: green;">Option green</div>'
      - Option 3
- id: 'properties.options.label: html'
  type: AutoComplete
  properties:
    options:
      - label: '<div>Some main text</div><div style="font-size: 6px;">Some small subtext</div>'
        value: 'one value'
      - label: '<div style="color: green;">Option green</div>'
        value: 'two value'
      - label: Option 3
        value: 'three'
- id: 'properties.options-style'
  type: AutoComplete
  properties:
    options:
      - label: one
        value: 'one value'
        style:
          color: red
      - label: two
        value: 'two value'
- id: 'properties.options.label: html and properties.options.filterString'
  type: AutoComplete
  properties:
    options:
      - label: '<div>Some main text</div><div style="font-size: 6px;">Some small subtext</div>'
        filterString: 'Some main text'
        value: 'one value'
      - label: '<div style="color: green;">Option green</div>'
        filterString: 'Option green'
        value: 'two value'
      - label: Option 3
        value: 'three value'
