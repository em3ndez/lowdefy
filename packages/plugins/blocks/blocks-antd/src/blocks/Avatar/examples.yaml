# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Avatar
- id: properties.color
  type: Avatar
  properties:
    color: '#f12345'
- id: properties.alt
  type: Avatar
  properties:
    alt: Alt text
- id: properties.content
  type: Avatar
  properties:
    content: Avatar
- id: properties.gap
  type: Avatar
  properties:
    content: Avatar
    gap: 40
# FIX: icon does not center vertically
- id: properties.icon string
  type: Avatar
  properties:
    icon: AiOutlineUser
- id: properties.icon object
  type: Avatar
  properties:
    icon:
      name: AiOutlineUser
      color: red
- id: properties.shape
  type: Avatar
  properties:
    shape: square
- id: properties.icon string properties.shape square
  type: Avatar
  properties:
    shape: square
    icon: AiOutlineUser
- id: properties.size small
  type: Avatar
  properties:
    size: small
- id: properties.size large
  type: Avatar
  properties:
    size: large
# version 4.8
# - id: properties.size 40
#   type: Avatar
#   properties:
#     size: 40
- id: properties.src
  type: Avatar
  properties:
    src: https://docs.lowdefy.com/images/favicon.png
- id: properties.content
  type: Avatar
  properties:
    content: Avatar
    style:
      border: 1px solid red
