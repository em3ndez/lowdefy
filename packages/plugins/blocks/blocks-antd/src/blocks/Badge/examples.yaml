# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Badge
- id: 'properties.count'
  type: Badge
  properties:
    count: 50
- id: 'properties.title: When hovering over badge'
  type: Badge
  properties:
    count: 50
    title: When hovering over badge
- id: 'properties.overflowCount'
  type: Badge
  properties:
    count: 15
    overflowCount: 10
- id: 'properties.count: 0'
  type: Badge
  properties:
    count: 0
- id: 'properties.showZero'
  type: Badge
  properties:
    count: 0
    showZero: true
- id: 'properties.status: success'
  type: Badge
  properties:
    status: success
# on antd but not working
- id: 'properties.status: processing'
  type: Badge
  properties:
    status: processing
- id: 'properties.status: default'
  type: Badge
  properties:
    status: default
- id: 'properties.status: error'
  type: Badge
  properties:
    status: error
- id: 'properties.status: error and text: Error!'
  type: Badge
  properties:
    status: error
    text: Error!
- id: 'properties.dot: true'
  type: Badge
  properties:
    dot: true
- id: 'properties.color'
  type: Badge
  properties:
    color: '#9c27b0'
- id: 'properties.offset [x,y]'
  type: Badge
  properties:
    dot: true
    offset:
      - 25
      - 10
- id: 'properties.icon string'
  type: Badge
  properties:
    icon: AiOutlineNotification
- id: 'properties.icon object'
  type: Badge
  properties:
    icon:
      name: AiOutlineNotification
      spin: true
- id: 'properties.size: small'
  type: Badge
  properties:
    size: 'small'
    count: 11
