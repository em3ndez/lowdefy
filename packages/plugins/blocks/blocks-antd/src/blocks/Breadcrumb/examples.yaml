# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Breadcrumb
- id: properties.list.label strings
  type: Breadcrumb
  properties:
    list:
      - Page One
      - Page two
- id: properties.list.pageId
  type: Breadcrumb
  properties:
    list:
      - pageId: page1
      - pageId: page2
- id: properties.list.icon
  type: Breadcrumb
  properties:
    list:
      - label: label 1
        icon: AiOutlineSave
      - label: label 2
        icon: AiOutlineDelete
- id: properties.list.icon with onClick
  type: Breadcrumb
  properties:
    list:
      - label: label 1
        icon: AiOutlineSave
      - label: label 2
        icon: AiOutlineDelete
  events:
    onClick: []
- id: properties.list.url
  type: Breadcrumb
  properties:
    list:
      - url: https://www.url.com
      - url: https://docs.url.com
- id: 'properties.list.pageId label and url'
  type: Breadcrumb
  properties:
    list:
      - url: https://www.url.com
        pageId: page1
      - url: https://www.url.com
        label: label url
      - pageId: page2
        label: label pageId
      - url: https://docs.url.com
        pageId: page3
      - label: label4
- id: properties.list.style
  type: Breadcrumb
  properties:
    list:
      - label: Page One
        style:
          color: blue
      - label: Page two
        style:
          color: red
- id: properties.separator
  type: Breadcrumb
  properties:
    separator: '->'
    list:
      - label: Page One
      - label: Page two
- id: 'properties.style: fontSize: 18'
  type: Breadcrumb
  properties:
    style:
      fontSize: 18
    list:
      - label: Page One
      - label: Page two
