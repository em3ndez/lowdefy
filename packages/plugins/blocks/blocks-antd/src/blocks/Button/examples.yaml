# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: Button
- id: properties.title
  type: Button
  properties:
    title: Button title
- id: properties.icon string
  type: Button
  properties:
    icon: AiOutlineSave
- id: properties.icon object
  type: Button
  properties:
    icon:
      name: AiOutlineSave
      color: red
- id: 'properties.shape: circle'
  type: Button
  properties:
    shape: circle
- id: 'properties.shape: round'
  type: Button
  properties:
    shape: round
- id: 'properties.block: true'
  type: Button
  properties:
    block: true
- id: 'properties.color: #9c27b0'
  type: Button
  properties:
    color: '#9c27b0'
- id: 'properties.size: small'
  type: Button
  properties:
    size: small
- id: 'properties.size: large'
  type: Button
  properties:
    size: large
- id: 'properties.danger: true'
  type: Button
  properties:
    danger: true
- id: 'properties.type: default'
  type: Button
  properties:
    type: default
- id: 'properties.type: dashed'
  type: Button
  properties:
    type: dashed
- id: 'properties.type: danger'
  type: Button
  properties:
    type: danger
- id: 'properties.type: link'
  type: Button
  properties:
    type: link
- id: 'properties.type: text'
  type: Button
  properties:
    type: text
- id: 'properties.disabled: true'
  type: Button
  properties:
    disabled: true
- id: 'properties.ghost: true'
  type: Button
  properties:
    ghost: true
- id: 'properties.hideTitle: true'
  type: Button
  properties:
    hideTitle: true
    icon: AiOutlineDelete
- id: 'properties.href: www.lowdefy.com'
  type: Button
  properties:
    href: https://www.lowdefy.com
- id: 'with html in title'
  type: Button
  properties:
    title: '<div style="color: orange;">Title</div>'
- id: 'with html in title and icon'
  type: Button
  properties:
    icon: AiOutlineSave
    title: '<div style="color: red;">Title</div>'
