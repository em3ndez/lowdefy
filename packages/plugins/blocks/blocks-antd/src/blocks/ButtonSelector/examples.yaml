# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: ButtonSelector
- id: properties.title
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    title: 'properties.title: My ButtonSelector Title'
- id: 'properties.options-string'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
- id: 'properties.options-number'
  type: ButtonSelector
  properties:
    options:
      - 15
      - 20
- id: 'properties.options-boolean'
  type: ButtonSelector
  properties:
    options:
      - true
      - false
- id: 'properties.options-object'
  type: ButtonSelector
  properties:
    options:
      - label: one
        value: 1
      - label: two
        value: 2
- id: 'properties.buttonStyle: outline'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    buttonStyle: outline
- id: 'properties.color_yellow' #use with buttonStyle: outline to see text on light colors
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    buttonStyle: outline
    color: yellow
- id: 'properties.color_red'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    color: red
- id: 'properties.disabled: true'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    disabled: true
- id: 'properties.size: small'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    size: small
- id: 'properties.size_large'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    size: large
- id: 'properties.label: span_12'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      span: 12
- id: 'properties.label: align_right'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      align: right
- id: 'properties.label: inline_true'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      inline: true
- id: 'properties.label.disabled: true'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      disabled: true
- id: 'properties.label.colon: false'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      colon: false
- id: 'properties.label.extra: showingextra'
  type: ButtonSelector
  properties:
    options:
      - Option 1
      - Option 2
    label:
      extra: showing extra
- id: 'properties.options: html'
  type: ButtonSelector
  properties:
    options:
      - '<div>Some main text</div><div style="font-size: 6px;">Some small subtext</div>'
      - '<div style="color: green;">Option green</div>'
      - Option 3
- id: 'properties.options.label: html'
  type: ButtonSelector
  properties:
    options:
      - label: '<div>Some main text</div><div style="font-size: 6px;">Some small subtext</div>'
        value: 1
      - label: '<div style="color: green;">Option green</div>'
        value: 2
      - label: Option 3
        value: 3
- id: 'properties.options-style'
  type: ButtonSelector
  properties:
    options:
      - label: one
        value: 1
        style:
          background: red
      - label: two
        value: 2
