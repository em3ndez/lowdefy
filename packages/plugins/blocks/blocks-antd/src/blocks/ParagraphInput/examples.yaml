# Copyright 2020-2024 Lowdefy, Inc

# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at

#     http://www.apache.org/licenses/LICENSE-2.0

# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

- id: default
  type: ParagraphInput
# - id: 'properties.code: true'
#   type: ParagraphInput
#   properties:
#     code: true
# - id: 'properties.delete: true'
#   type: ParagraphInput
#   properties:
#     delete: true
# - id: 'properties.disabled: true'
#   type: ParagraphInput
#   properties:
#     disabled: true
# - id: 'properties.mark: true'
#   type: ParagraphInput
#   properties:
#     mark: true
# - id: 'properties.strong: true'
#   type: ParagraphInput
#   properties:
#     strong: true
# - id: 'properties.italic: true'
#   type: ParagraphInput
#   properties:
#     italic: true
# - id: 'properties.underline: true'
#   type: ParagraphInput
#   properties:
#     underline: true
# - id: 'properties.type: secondary'
#   type: ParagraphInput
#   properties:
#     type: secondary
# - id: 'properties.type: warning'
#   type: ParagraphInput
#   properties:
#     type: warning
# - id: 'properties.type: danger'
#   type: ParagraphInput
#   properties:
#     type: danger
# - id: 'properties.copyable: true'
#   type: ParagraphInput
#   properties:
#     copyable: true
# - id: 'properties.copyable: text: one'
#   type: ParagraphInput
#   properties:
#     copyable:
#       text: one
# - id: 'properties.copyable: icon'
#   type: ParagraphInput
#   properties:
#     copyable:
#       icon: AiOutlineWarning
# - id: 'properties.copyable: icon: []'
#   type: ParagraphInput
#   properties:
#     copyable:
#       icon:
#         - AiOutlineWarning
#         - AiOutlineRocket
# - id: 'properties.copyable: tooltips: []'
#   type: ParagraphInput
#   properties:
#     copyable:
#       tooltips:
#         - Click to copy
#         - Copy done
# - id: 'properties.ellipsis: true'
#   type: ParagraphInput
#   style:
#     width: 300
#   properties:
#     ellipsis: true
# - id: 'properties.ellipsis: rows'
#   type: ParagraphInput
#   style:
#     width: 300
#   properties:
#     ellipsis:
#       rows: 3
# - id: 'properties.ellipsis: expandable, rows'
#   type: ParagraphInput
#   style:
#     width: 300
#   properties:
#     ellipsis:
#       rows: 3
#       expandable: true
# - id: 'properties.ellipsis: rows, suffix'
#   type: ParagraphInput
#   style:
#     width: 300
#   properties:
#     ellipsis:
#       rows: 3
#       suffix: and other
# # FIX Jest: not working, might be and antd issue.
# # - id: 'properties.ellipsis: rows, symbol'
# #   type: ParagraphInput
# #   style:
# #     width: 300
# #   properties:
# #     ellipsis:
# #       symbol: '???'

# - id: 'properties.editable: true'
#   type: ParagraphInput
#   properties:
#     editable: true
# - id: 'properties.editable: false'
#   type: ParagraphInput
#   properties:
#     editable: false
# - id: 'properties.editable.icon'
#   type: ParagraphInput
#   properties:
#     editable:
#       icon: AiOutlinePlus
# - id: 'properties.editable.tooltip'
#   type: ParagraphInput
#   properties:
#     editable:
#       tooltip: Change the value
# # FIX: TypeError: Cannot read property 'focus' of null
# # - id: 'properties.editable.editing: true'
# #   type: ParagraphInput
# #   properties:
# #     editable:
# #       editing: true
# - id: 'properties.editable.maxLength: 10'
#   type: ParagraphInput
#   properties:
#     editable:
#       maxLength: 10
# # - id: 'properties.editable.autoSize: true'
# #   type: ParagraphInput
# #   properties:
# #     editable:
# #       autoSize: true
# # - id: 'properties.editable.autoSize: min max'
# #   type: ParagraphInput
# #   properties:
# #     editable:
# #       autoSize:
# #         minRows: 2
# #         maxRows: 4
