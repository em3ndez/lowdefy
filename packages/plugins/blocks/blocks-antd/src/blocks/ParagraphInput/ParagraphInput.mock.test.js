/*
  Copyright 2020-2024 Lowdefy, Inc

  Licensed under the Apache License, Version 2.0 (the "License");
  you may not use this file except in compliance with the License.
  You may obtain a copy of the License at

      http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing, software
  distributed under the License is distributed on an "AS IS" BASIS,
  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  See the License for the specific language governing permissions and
  limitations under the License.
*/

import { runMockRenderTests } from '@lowdefy/block-dev';

import Block from './ParagraphInput.js';
import examples from './examples.yaml';
import schema from './schema.json';

const testConfig = {
  required: true,
  validation: true,
  values: [
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
  ],
};

jest.mock('antd', () => {
  const comp = jest.fn(() => 'mocked');
  comp.Paragraph = jest.fn(() => 'mocked');
  return {
    Typography: comp,
  };
});
const mocks = [
  {
    getMockFns: async () => {
      const antd = await import('antd');
      return [antd.Typography.Paragraph];
    },
    getBlock: async () => {
      const Block = await import('./ParagraphInput.js');
      return Block.default;
    },
    name: 'ParagraphInput',
  },
];
runMockRenderTests({ Block, examples, mocks, schema, testConfig });
