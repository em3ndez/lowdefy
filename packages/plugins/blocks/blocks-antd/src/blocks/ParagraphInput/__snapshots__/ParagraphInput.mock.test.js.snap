// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock render - default - value[0] - ParagraphInput 1`] = `mocked`;

exports[`Mock render - default - value[0] - ParagraphInput 2`] = `
Array [
  Array [
    Object {
      "children": "",
      "className": "{}",
      "code": undefined,
      "copyable": undefined,
      "delete": undefined,
      "disabled": undefined,
      "editable": Object {
        "onChange": [Function],
        "onStart": [Function],
      },
      "ellipsis": undefined,
      "id": "default",
      "italic": undefined,
      "mark": undefined,
      "strong": undefined,
      "type": undefined,
      "underline": undefined,
    },
    Object {},
  ],
]
`;

exports[`Mock render - default - value[1] - ParagraphInput 1`] = `mocked`;

exports[`Mock render - default - value[1] - ParagraphInput 2`] = `
Array [
  Array [
    Object {
      "children": "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
      "className": "{}",
      "code": undefined,
      "copyable": undefined,
      "delete": undefined,
      "disabled": undefined,
      "editable": Object {
        "onChange": [Function],
        "onStart": [Function],
      },
      "ellipsis": undefined,
      "id": "default",
      "italic": undefined,
      "mark": undefined,
      "strong": undefined,
      "type": undefined,
      "underline": undefined,
    },
    Object {},
  ],
]
`;
